.eslintrc

```diff
@@ -101,5 +101,43 @@
     },
     "parserOptions": {
         "ecmaVersion": 2018
-    }
+    },
+    "overrides": [
+        {
+            "files": ["test/**"],
+            "env": {
+                "mocha": true,
+                "es2020": true
+            },
+            "plugins": [
+                "mocha"
+            ],
+            "extends": [
+                "plugin:mocha/recommended"
+            ],
+            "rules": {
+                "mocha/no-hooks-for-single-case": "off",
+                "mocha/no-setup-in-describe": "off",
+                "camelcase": "off",
+                "max-len": ["warn", 130],
```

Comment:
There are some configurations defined for them above. Can we use the settings above to simplify this a bit more?

Reply:
What do you mean? We don't use them since we have different values (130 for line length, root config has 120) for test folder.

Do you mean we should have the same values?

Reply:
I mean they are not hugely different, if there is no special reason for the difference, I would just use what is defined on the top level

Reply:
Due to test names and nesting, line length is generally longer in test files. I think it can stay like this. Otherwise the code will be ugly with line breaks in test names.

Regarding "mocha/" rules they come from plugin so we cannot add them to root rules.

camelcase: "off" exists since there are some an_integer like variable names in a serialization test I suppose.

and the "padded-blocks" exists only on test files since recently I wanted some consistency about it. I think enabling it for typescript files is not too necessary since I don't feel files look inconsistent in style. This is probably due to every file has different structure whereas every test file has the describe block.

 I think making overrides shorter is not a simplification, it restricts out ability to choose rules for each glob pattern